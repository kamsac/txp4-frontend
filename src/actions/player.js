import dotenvConfiguration from '../dotenv-configuration';
import PlayerResourceMock from '../resources/player/mock';
import PlayerResource from '../resources/player';

export const RECEIVE_PLAYER = 'player:receivePlayer';
export function receivePlayer(player) {
  return {
    type: RECEIVE_PLAYER,
    payload: { player },
  };
}

function fetchPlayer() {
  const resource = dotenvConfiguration.API_URL ? PlayerResource : PlayerResourceMock;

  return (dispatch) => {
    resource.getPlayer()
      .then((response) => {
        dispatch(receivePlayer(response.data));
      });
  };
}

export const REQUEST_PLAYER = 'player:requestPlayer';
export function requestPlayer() {
  return (dispatch) => {
    dispatch(fetchPlayer());
  };
}
