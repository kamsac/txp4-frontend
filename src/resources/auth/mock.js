export const ACCESS_TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjoxNTE2MjM5MDIyLCJsb2dpbiI6Imx5bWFrIiwibmljayI6IiRzJG8kYTYwbCRjODJ5JGZhM20kYzgyYSRhNjBrIn0.aaKzXEZP2tovhBAqPJYBdmbLzKjfdBsuX3FaB58JZvE';

const AuthResourceMock = {
  login() {
    return new Promise((resolve) => {
      const response = {
        data: {
          token: ACCESS_TOKEN,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default AuthResourceMock;
