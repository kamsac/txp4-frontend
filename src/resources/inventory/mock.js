const ITEMS = [
  {
    id: "a1a1a1",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "b2b2b2",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "c3c3c3",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "d4d4d4",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "e5e5e5",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "f6f6f6",
    name: "Duper engine",
    vendorId: 'aaa',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1.5,
      },
      {
        name: "acceleration",
        value: 3,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "f7f7f7",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "g8g8g8",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "h9h9h9",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "i0i0i0",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "j1j1j1",
    name: "Transmission",
    vendorId: 'ccc',
    equipRegion: "transmission",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.7,
      },
      {
        name: "acceleration",
        value: 1.5,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
];

const InventoryResourceMock = {
  getItems() {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem(item) {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(ITEMS), 100);
    });
  },
};

export default InventoryResourceMock;
