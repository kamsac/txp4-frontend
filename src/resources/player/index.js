import axios from 'axios';
import dotenvConfiguration from '../../dotenv-configuration';

const PlayerResource = {
  getPlayer(playerLogin) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}`);
  },

  getInventory(playerLogin) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory`);
  },

  equipItem(playerLogin, itemId) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory/${itemId}/equip`);
  },

  unequipItem(playerLogin, itemId) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory/${itemId}/unequip`);
  },
};

export default PlayerResource;
