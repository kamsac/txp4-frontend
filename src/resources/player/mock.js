export const ITEMS = [
  {
    id: 'a1a1a1',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'b2b2b2',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'c3c3c3',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'd4d4d4',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'e5e5e5',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'f6f6f6',
    name: 'Duper engine',
    vendorId: 'aaa',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1.5,
      },
      {
        name: 'acceleration',
        value: 3,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'f7f7f7',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'g8g8g8',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'h9h9h9',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'i0i0i0',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'j1j1j1',
    name: 'Transmission',
    vendorId: 'ccc',
    equipRegion: 'transmission',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.7,
      },
      {
        name: 'acceleration',
        value: 1.5,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
];

export const PLAYER = {
  login: 'lymak',
  nick: '$s$o$a60l$c82y$fa3m$c82a$a60k',
  inventory: {
    items: ITEMS,
  },
};

const PlayerResourceMock = {
  getPlayer() {
    return new Promise((resolve) => {
      const response = {
        data: PLAYER,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  getInventory() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  unequipItem() {
    return new Promise((resolve) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default PlayerResourceMock;
