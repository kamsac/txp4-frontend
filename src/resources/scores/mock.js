const SCORES = [
  {
    player: {
      login: 'lymak',
      nickname: 'lymak',
    },
    points: 4242,
  },
  {
    player: {
      login: 'kibes',
      nickname: 'kibes',
    },
    points: 3232,
  },
  {
    player: {
      login: 'tomek',
      nickname: 'tomek',
    },
    points: 2416,
  },
  {
    player: {
      login: 'omed',
      nickname: 'omed',
    },
    points: 2000,
  },
  {
    player: {
      login: 'tar',
      nickname: 'tar',
    },
    points: 1960,
  },
  {
    player: {
      login: 'oiz',
      nickname: 'oiz',
    },
    points: 680,
  },
  {
    player: {
      login: 'xetrow',
      nickname: 'xetrow',
    },
    points: 403,
  },
  {
    player: {
      login: 'ginok',
      nickname: 'ginok',
    },
    points: 30,
  },
  {
    player: {
      login: 'llif',
      nickname: 'llif',
    },
    points: 20,
  },
  {
    player: {
      login: 'ida',
      nickname: 'ida',
    },
    points: 10,
  },
];

const ScoresResourceMock = {
  getScores() {
    return new Promise((resolve, reject) => {
      const response = {
        data: SCORES,
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default ScoresResourceMock;