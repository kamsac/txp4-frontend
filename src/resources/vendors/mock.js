const VENDORS = [
  {
    id: 'aaa',
    name: "Super",
    color: "#8f0",
  },
  {
    id: 'bbb',
    name: "Safe",
    color: "#f80",
  },
  {
    id: 'ccc',
    name: "Heavy",
    color: "#800",
  },
  {
    id: 'ddd',
    name: "Light",
    color: "#08f",
  },
];

const VendorsResourceMock = {
  getVendors() {
    return new Promise((resolve, reject) => {
      const response = {
        data: VENDORS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default VendorsResourceMock;
