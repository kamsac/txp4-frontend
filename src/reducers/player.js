import inventoryReducer from './inventory';
import { RECEIVE_PLAYER } from '../actions/player';
import { EQUIP_ITEM, RECEIVE_INVENTORY, REQUEST_INVENTORY } from '../actions/inventory';

const initialState = {
  login: '',
  nick: '',
  inventory: {
    items: [],
  },
};

export default function playerReducer(state = initialState, action) {
  switch (action.type) {
    case RECEIVE_PLAYER:
      return Object.assign({}, state, action.payload.player);

    case REQUEST_INVENTORY:
    case RECEIVE_INVENTORY:
    case EQUIP_ITEM: {
      const inventory = inventoryReducer(state.inventory, action);
      return Object.assign({}, state, { inventory });
    }
    default:
      return state;
  }
}
