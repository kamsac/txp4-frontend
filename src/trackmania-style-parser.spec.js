import trackmaniaStyleParser from './trackmania-style-parser';
import { PLAYER } from './resources/player/mock';

describe('trackmaniaStyleParser', () => {
  it('should parse', () => {
    expect(trackmaniaStyleParser(PLAYER.nick)).toMatchSnapshot();
  });

  it('should strip links', () => {
    const nickWithLink = `$l[https://example.com]${PLAYER.nick}`;
    const parsedNick = trackmaniaStyleParser(nickWithLink);
    expect(parsedNick).not.toContain('<a');
    expect(parsedNick).not.toContain('href');
  });
});
