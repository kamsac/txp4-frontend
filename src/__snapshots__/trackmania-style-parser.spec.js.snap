// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trackmaniaStyleParser should parse 1`] = `"<span style=\\"font-weight:bold;color:#a60;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">l</span><span style=\\"font-weight:bold;color:#c82;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">y</span><span style=\\"font-weight:bold;color:#fa3;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">m</span><span style=\\"font-weight:bold;color:#c82;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">a</span><span style=\\"font-weight:bold;color:#a60;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">k</span>"`;
